server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/ParquetResolver.java

```diff
@@ -19,52 +19,63 @@
  * under the License.
  */
 
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.example.data.simple.NanoTime;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
 import org.greenplum.pxf.api.OneField;
 import org.greenplum.pxf.api.OneRow;
-import org.greenplum.pxf.api.model.Resolver;
 import org.greenplum.pxf.api.UnsupportedTypeException;
 import org.greenplum.pxf.api.io.DataType;
 import org.greenplum.pxf.api.model.RequestContext;
-import org.greenplum.pxf.api.model.BasePlugin;
-
-import org.greenplum.pxf.plugins.hdfs.utilities.HdfsUtilities;
-import org.apache.parquet.example.data.Group;
-import org.apache.parquet.schema.MessageType;
-import org.apache.parquet.schema.OriginalType;
-import org.apache.parquet.schema.PrimitiveType;
-import org.apache.parquet.schema.Type;
+import org.greenplum.pxf.api.model.Resolver;
 
+import java.io.IOException;
 import java.math.BigDecimal;
 import java.math.BigInteger;
 import java.nio.ByteBuffer;
 import java.sql.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 import java.util.LinkedList;
 import java.util.List;
 
-public class ParquetResolver extends BasePlugin implements Resolver {
+public class ParquetResolver extends ParquetFileAccessor implements Resolver {
 
-    public static final int JULIAN_EPOCH_OFFSET_DAYS = 2440588;
-    public static final long MILLIS_IN_DAY = 24 * 3600 * 1000;
+    private static final int JULIAN_EPOCH_OFFSET_DAYS = 2440588;
+    private static final long MILLIS_IN_DAY = 24 * 3600 * 1000;
+    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
 
-    /**
-     * {@inheritDoc}
-     * @param schema the MessageType instance, which is obtained from the Parquet file footer
-     */
-    public List<OneField> getFields(OneRow row, MessageType schema) throws Exception
-    {
-      ParquetUserData parquetUserData = new ParquetUserData(schema);
-      Group g = (Group) row.getData();
-      List<OneField> output = resolveRecord(parquetUserData, g);
-      return output;
+    private static Date epoch;
+    static {
+        try {
+            epoch =  dateFormat.parse("1970-01-01 00:00:00");
```

Comment:
Another approach would be to hardcode the epoch value, and get rid of the static initialization block.
```suggestion
            epoch =  new Date( 1081157732000L );
```
However, I am not sure how this will play out with different timezones